### PR TITLE
Only copy parts of the schedule when collapsing final const-true specialization

### PR DESCRIPTION
**Pull request by psuriana**

### CODE REVIEW

**Comment by steven-johnson** [comment]
Yeah, it may fix the issue; I'm just concerned that the optimization may be too fragile to justify itself: it's not necessary for correctness at all, it just avoid some later work in simplify(). 

That said: If we're confident that this fix will be robust and maintainable, LGTM.

**Comment by abadams** [comment]
Longer term fix is split Schedule into two classes - the parts that apply individually to each Stage (splits, vars), and the parts that apply to an entire Func (compute_at, bound, storage_dims). Specializations only get the Stage stuff.

**Comment by steven-johnson** [comment]
We should probably add a test case -- the added clause in specialize.cpp in PR #1998 would probably suffice.

**Comment by steven-johnson** [comment]
So what's the verdict: are we going to land this change, or just back out the optimization entirely? (I'm still a bit concerned that it's an optimization that may not be valuable enough to outweigh its possible fragility, but if abadams@ and psuriana@ are happy with it, LGTM.)

File: test/correctness/specialize.cpp [hunk 1]
```diff
@@ -609,7 +609,33 @@ int main(int argc, char **argv) {
...
+        Param<int> p;
+        Expr const_true = Expr(0) == Expr(0);
+        Expr different_const_true = Expr(1) == Expr(1);
+
+        // Check that if we promote a final const-true specialize, we keep the
+        // implicit compute/store_root required for outputs.
+        Func f("foof");
+        f(x) = x;
+        f.specialize(p == 0).vectorize(x, 32);      // will *not* be pruned
+        f.specialize(const_true);                   // dupe of call above, won't add new specialization
```
Comment (abadams) [inline test/correctness/specialize.cpp]: Should vectorize by 16 in this case and check for it below so that it can be distinguished from the base case

**Comment by steven-johnson** [comment]
Any objections to merging?

**Pull request merged**